apps/storefront-e2e/src/integration/search.cy.ts

```diff
@@ -6,76 +6,81 @@ import {
+        searchPage.getFacets().resetRating();
+
+        // apply 1st filter
+        searchPage.getFacets().setLabel('New');
+        searchPage.waitForSearchRequest();
+        checkProductCardsFilteringByName(searchPage, 4, 2, query);
+
+        // apply 2nd filter
+        searchPage.getFacets().setColor('Black');
+        searchPage.waitForSearchRequest();
+        checkProductCardsFilteringByName(searchPage, 4, 1, query);
+
+        // clear 2nd filter
+        // we don't expect search request here because previous query is cached
+        searchPage.getFacets().resetColor();
+        checkProductCardsFilteringByName(searchPage, 4, 2, query);
+      });
     });
 
-    it('should apply price filterring', () => {
-      // set brand to limit a number of products displayed
-      // and avoid issues with concrete products displaying
-      searchPage = new SearchPage({ q: 'Cameras' });
-      searchPage.visit();
-      searchPage.getFacets().setBrand('Kodak');
-
+    describe('by price', () => {
       const minPrice = 200;
       const maxPrice = 400;
 
-      cy.log('set minimum price');
-      searchPage.getFacets().setMinPrice(minPrice);
-      searchPage.waitForSearchRequest();
-      checkProductCardsFilteringByPrice(searchPage, minPrice);
-
-      cy.log('reset prices, set max price');
-      searchPage.getFacets().resetPrices();
-      searchPage.waitForSearchRequest();
-      searchPage.getFacets().setMaxPrice(maxPrice);
-      searchPage.waitForSearchRequest();
-      checkProductCardsFilteringByPrice(searchPage, 0, maxPrice);
-
-      cy.log('reset prices, change min price range');
-      searchPage.getFacets().resetPrices();
-      searchPage.waitForSearchRequest();
-      searchPage.getFacets().setMinPriceRange(minPrice);
-      searchPage.waitForSearchRequest();
-      checkProductCardsFilteringByPrice(searchPage, minPrice);
-
-      cy.log('reset prices, change max price range');
-      searchPage.getFacets().resetPrices();
-      searchPage.waitForSearchRequest();
-      searchPage.getFacets().setMaxPriceRange(maxPrice);
-      searchPage.waitForSearchRequest();
-      checkProductCardsFilteringByPrice(searchPage, 0, maxPrice);
+      beforeEach(() => {
+        // set brand to limit a number of products displayed
+        // and avoid issues with concrete products displaying
+        searchPage = new SearchPage({ q: 'Cameras', search: 'brand=Kodak' });
+        searchPage.visit();
+      });
+
+      it('should apply price filtering', () => {
+        cy.log('set minimum price');
+        searchPage.getFacets().setMinPrice(minPrice);
+        searchPage.waitForSearchRequest();
+        checkProductCardsFilteringByPrice(searchPage, minPrice);
+
+        cy.log('reset prices, set max price');
+        searchPage.getFacets().resetPrices();
+        searchPage.waitForSearchRequest(true);
+        searchPage.getFacets().setMaxPrice(maxPrice);
+        searchPage.waitForSearchRequest();
```

Comment:
I don't understand the logic behind skipping / not skipping the requests here - in some places we have it skipped, in other it still appears

Can you explain it, please? 

---
<img width="424" alt="Screenshot 2023-11-07 at 08 28 03" src="https://github.com/spryker/oryx/assets/14858875/b56ca10d-ad88-441f-b360-c94de6158331">

Reply:
I also can't understand how skipping an API call wait increases test stability. 

In theory, it will make them unstable because API calls are random

Reply:
Because we are not making some request (they have been already cached), but UI still need some time to re-build the template